README.md

```diff
@@ -1,12 +1,78 @@
 
-## Contexts
-In Kinase, content mapped to the provided annotations is grouped together in a context. This context is keyed by an arbitrary string in the redux state, and might represent a specific user tagging content in the extension or a project that content is being tagged for.
 
-To switch contexts from another extension or webpage (e.g., if a user first reads an instruction page, you can set the project context from there), you can send a message to the extension with:
+## Concepts
+### Annotations
+Annotations represent the labels which you’d like to select content for. An annotation is represented by a string identifier (e.g., `products`) and a schema containing the fields and their types that can be mapped to the annotation (e.g., `name`, `description`, `photo`). The schema can optionally specify the `multiple` flag to indicate that multiple groupings of these fields can be mapped to the annotation.
+
+NB: Currently only `text`, `rich-text`, and `image` fields are supported. Help us out and add more!
+
+```JSON
+{
+  "products": {
+    "multiple": true,
+    "fields": {
+      "name": "text",
+      "description": "rich-text",
+      "photo": "image"
+    }
+  }
+}
+```
+
+### Mappings
+A mapping represents selected content for a given annotation. If `multiple` is specified in its schema, the annotation can have multiple mappings (though the mappings for an annotation will be in array form, without the `multiple` flag the array will always be of length 1).
+
+Each mapped field in a mapping contains the content taken from the website (with any additional user edits) and the original source of that content (the URL of the website it was taken from, and a unique CSS selector specifying its container).
+
+The mappings for our `products` annotation might look like this:
+```JSON
+{
+  "products": [{
+    "name": {
+      "content": "Laptop 3000",
+      "sources": [{
+        "url": "https://laptop-world.com",
+        "selector": ".laptops:nth-child(0) > h3"
+      }]
+    },
+    "description": {
+      "content": "<p>Our <em>top of the line</em> model, now yours for only $9999.99!</p>",
+      "sources": [{
+        "url": "https://laptop-world.com",
+        "selector": ".laptops:nth-child(0) > .description"
+      }]
+    },
+    "photo": {
+      "content": "https://laptop-world.com/laptop-3000.jpg",
+      "sources": [{
+        "url": "https://laptop-world.com",
+        "selector": ".laptops:nth-child(0) > .photo"
+      }]
+    }
+  }]
+}
+```
+
+### Contexts
+In Kinase, annotations and their mappings are grouped together in a context. This context is keyed by an arbitrary string in the redux state, and might represent a specific user tagging content in the extension or a project that content is being tagged for.
 
+```JSON
+{
+  "laptop-world": {
```

Comment:
laptop-world is confusing because it was also a source---i can't think of a better example off the top of my mind...